indico/modules/events/papers/client/js/components/TimelineItem.jsx

```diff
@@ -0,0 +1,117 @@
+// This file is part of Indico.
+// Copyright (C) 2002 - 2019 CERN
+//
+// Indico is free software; you can redistribute it and/or
+// modify it under the terms of the MIT License; see the
+// LICENSE file for more details.
+
+import _ from 'lodash';
+import React, {useState} from 'react';
+import PropTypes from 'prop-types';
+import moment from 'moment';
+import {useSelector} from 'react-redux';
+import {Transition} from 'semantic-ui-react';
+
+import {Param, Translate} from 'indico/react/i18n';
+import {serializeDate} from 'indico/utils/date';
+
+import PaperReviewForm from './PaperReviewForm';
+import RevisionJudgment from './RevisionJudgment';
+import RevisionTimeline from './RevisionTimeline';
+import SubmitRevision from './SubmitRevision';
+import UserAvatar from './UserAvatar';
+import {canCommentPaper, canReviewPaper, getPaperDetails} from '../selectors';
+
+export default function TimelineItem({revision}) {
+  const {submitter, isLastRevision, number, submittedDt, files, timeline} = revision;
+  const submitterName = submitter.isSystem ? Translate.string('A user') : submitter.fullName;
+  const {isInFinalState} = useSelector(getPaperDetails);
+  const canComment = useSelector(canCommentPaper);
+  const canReview = useSelector(canReviewPaper);
+  const [visible, setVisible] = useState(false);
+
+  return (
+    <>
+      <div className="i-timeline">
+        <div className="i-timeline-item">
+          <UserAvatar user={submitter} />
+          <div
+            className={`i-timeline-item-box header-indicator-left ${
+              !isLastRevision && !visible ? 'header-only' : ''
+            }`}
+            id={`revision-info-${revision.id}`}
```

Comment:
do we still need that id?

Reply:
Yeah, link inside `PaperMetadata` allows to quickly jump to a specific revision.